app/src/main/java/com/lightricks/feedexercise/ui/feed/Error.kt

```diff
@@ -0,0 +1,28 @@
+package com.lightricks.feedexercise.ui.feed
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.lightricks.feedexercise.R
+import com.lightricks.feedexercise.ui.feed.model.ViewState
+
+@Composable
+internal fun ErrorState(
+    state: ViewState.Error,
+    viewModel: FeedViewModel
```

Comment:
I have never worked with compose but I don't think it is a good idea to couple the component with the view model.

Reply:
You are right - it should be callback